write.go

```diff
@@ -177,8 +169,10 @@ func (w *Writer) writeField(field interface{}) error {
 		return w.writeLiteral(field)
 	case []interface{}:
 		return w.writeList(field)
+	case *timeLayout:
```

Comment:
How about something like this? Fewer conditionals, can just thread whatever you need through.

	// A time.Time with a specific layout
	type TimeLayout struct {
		time.Time
		layout string
	}

	func (d *TimeLayout) SetLayout(layout string) {
		d.layout = layout
	}

	func (d *TimeLayout) Parse(value string) (time.Time, error) {
		return time.Parse(d.layout, value)
	}

	func (d *TimeLayout) FormatLayout() string {
		return d.Format(d.layout)
	}